docs/modules/reference/pages/architecture/opennms_integration.adoc

```diff
@@ -0,0 +1,80 @@
+
+:imagesdir: ../assets/images
+= OpenNMS Integration
+
+Various components are involved in the lifecycle of an alarm and the correlation process between OpenNMS and ALEC:
+
+.Alarm lifecycle architecture diagram
+image::architecture/opennms_alarm_lifecycle.png[Architecture diagram that depicts the components involved in an alarm's lifecycle, 600]
+
+Events are generated by existing OpenNMS services that either process unsolicited messages or actively poll monitored network inventory.
+They are then sent to https://docs.opennms.com/horizon/latest/reference/daemons/daemon-config-files/eventd.html[eventd], where they are processed and enriched with the corresponding configuration data.
+
+https://docs.opennms.com/horizon/latest/reference/daemons/daemon-config-files/alarmd.html[Alarmd] listens for events that contain alarm meta-data and, when it identifies them, creates or updates corresponding alarms.
+It also exposes a series of event hooks that can be used to alter or enrich alarms when they are created.
+ALEC leverages these hooks to perform additional data processing and tag alarms with a managed object type and managed object instance (see <<managed_object_tagging>>).
+OpenNMS also allows for notifications to be generated when alarms are modified in any way.
+
+The xref:datasources/direct.adoc[direct datasource] leverages API hooks within the runtime, whereas the xref:datasources/kafka.adoc[Kafka datasource] leverages the Kafka integration to interface externally.
+
+[[managed_object_tagging]]
+== Managed object tagging
+
+Managed object tagging further contextualizes alarms by associating them with a specific component instance or inventory object in ALEC.
+The process is flexible and configurable, allowing alarms to be associated with components beyond the existing node, interface, service model that is built in to OpenNMS.
```

Comment:
```suggestion
The process is flexible and configurable, allowing alarms to be associated with components beyond the existing node, interface, service model that is built-in to OpenNMS.
```

Reply:
For some reason, the hyphen doesn't read smoothly to me. Maybe "built into"?

```suggestion
The process is flexible and configurable, allowing alarms to be associated with components beyond the existing node, interface, service model that is built into OpenNMS.
```

Reply:
Yes, that works.